java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -551,4 +556,1697 @@ public void runPiranhaWithFewProperties() throws IOException {
+    b.putFlag("Piranha:Config", "src/test/resources/config/properties_test_noFlag.json");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+          BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  if (experimentation.isToggleEnabled()) { int a = 1; }",
+              "     else { int b = 2;}",
+              "  if (experimentation.isUnrelatedToggleEnabled()) { int c = 1; }",
+              "     else { int d = 2;}",
+              "  if (experimentation.isToggleDisabled()) { return \"X\"; }",
+              "     else { return \"Y\";}",
+              " }",
+              "}")
+          .addOutputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  int a = 1;",
+              "  if (experimentation.isUnrelatedToggleEnabled()) { int c = 1; }",
+              "     else { int d = 2;}",
+              "  return \"Y\";",
+              " }",
+              "}");
+
+      bcr.doTest();
+    } catch (ParseException pe) {
+      pe.printStackTrace();
+      assert_().fail("Incorrect parameters passed to the checker");
+    }
+  }
+
+  /*
+   * This test passes flag name as a Piranha argument.
+   * Whether or not flag name is passed, the noFlagCase simplification should not be affected.
+   * Uses "properties_test_noFlag.json" as the config file.
+   * Note that "Piranha:FlagName" is not considered when argumentIndex is not specified.
+   * */
+  @Test
+  public void noFlagCaseWithFlagSpecified() throws IOException {
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "STALE_FLAG");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "src/test/resources/config/properties_test_noFlag.json");
+
+    try {
+      BugCheckerRefactoringTestHelper bcr =
+          BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+      bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+      bcr = addHelperClasses(bcr);
+      bcr.addInputLines(
+              "XPFlagCleanerSinglePositiveCase.java",
+              "package com.uber.piranha;",
+              "class XPFlagCleanerSinglePositiveCase {",
+              " private XPTest experimentation;",
+              " public String evaluate() {",
+              "  if (experimentation.isToggleEnabled()) { int a = 1; }",
```

Comment:
What does this code do about `experimentation.isToggleEnabled(SOME_FLAG) { ... }`?

Reply:
And as you know by now, argumentIndex not being specified would lead to all occurrences of `isToggleEnabled`, irrespective of its argument(s) count/value.

I'm in the process of adding `ArgumentIndexOptional` Piranha argument to fix this